src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/CertificateUtilityWin.cs

```diff
@@ -19,27 +21,58 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted
     [PlatformSpecific(TestPlatforms.Windows)]
     class CertificateUtilityWin
     {
+
+        public class CertificateOption
+        {
+            public string Subject { get; set; }
+            public string KeyExportPolicy { get; set; } = "Exportable";
+            public string CertificateStoreLocation { get; set; }
+            public List<Tuple<string, string, string>> TextExtensions { get; set; } = new List<Tuple<string, string, string>> { new Tuple<string, string, string>("2.5.29.3", "1.3.6.1.5.5.8.2.2", "1.3.6.1.4.1.311.10.3.11") };
+            public string KeySpec { get; set; } = "KeyExchange";
+            public string Provider { get; set; } = "Microsoft Enhanced RSA and AES Cryptographic Provider";
+            public string Type { get; set; } = "24";
+            public string KeyAlgorithm { get; set; } = "rsa";
+            public string KeyLength { get; set; } = "2048";
+            public string HashAlgorithm { get; set; } = "sha256";
+            public string KeyUsage { get; set; } = "None";
+            public string TestRoot { get; set; } = "-TestRoot";
+        }
+
         private CertificateUtilityWin()
         {
         }
 
         /// <summary>
         /// Create a self-signed certificate through makecert.
         /// </summary>
-        internal static void CreateCertificate(string certificateName, string certificateLocation, string providerName, string providerType)
+        internal static void CreateCertificate(CertificateOption option)
         {
-            Assert.False(string.IsNullOrWhiteSpace(certificateName), "FAILED: certificateName should not be null or empty.");
-            Assert.False(string.IsNullOrWhiteSpace(certificateLocation), "FAILED: certificateLocation should not be null or empty.");
-
-            string makecertPath = string.IsNullOrEmpty(DataTestUtility.MakecertPath) ? "makecert" : DataTestUtility.MakecertPath;
-            ProcessStartInfo processStartInfo = new ProcessStartInfo(makecertPath);
-            processStartInfo.Arguments = string.Format(@"-n ""CN={0}"" -pe -sr {1} -r -eku 1.3.6.1.5.5.8.2.2,1.3.6.1.4.1.311.10.3.11 -ss my -sky exchange -sp ""{2}"" -sy {3} -len 2048 -a sha256",
-                                        certificateName,
-                                        certificateLocation,
-                                        providerName,
-                                        providerType);
-            Process process = new Process();
-            process.StartInfo = processStartInfo;
+            Assert.False(string.IsNullOrWhiteSpace(option.Subject), "FAILED: certificateName should not be null or empty.");
+            Assert.False(string.IsNullOrWhiteSpace(option.CertificateStoreLocation), "FAILED: certificateLocation should not be null or empty.");
+
+
+
+            string powerShellPath = string.IsNullOrEmpty(DataTestUtility.MakecertPath) ? "powershell" : DataTestUtility.MakecertPath;
```

Comment:
I'm wondering why this command has been included inside the code from the beginning! Why don't we just add it to the pipeline and taking it out of the source code?